diff -urN a/rs/extensions.bzl b/rs/extensions.bzl
--- a/rs/extensions.bzl	2026-03-21 22:53:52
+++ b/rs/extensions.bzl	2026-03-21 22:54:11
@@ -816,9 +816,8 @@
 
     hub_contents = []
     for name, versions in versions_by_name.items():
-        binaries = annotation_for(annotations, name, version).gen_binaries
-
         for version in versions:
+            annotation = annotation_for(annotations, name, version)
             spoke_repo = _spoke_repo(hub_name, name, version)
 
             hub_contents.append("""
@@ -827,16 +826,30 @@
     actual = "@{spoke_repo}//:{name}",
 )""".format(name = name, version = version, spoke_repo = spoke_repo))
 
-            for binary in binaries:
+            for binary in annotation.gen_binaries:
                 hub_contents.append("""
 alias(
     name = "{name}-{version}__{binary}",
     actual = "@{spoke_repo}//:{binary}__bin",
 )""".format(name = name, version = version, binary = binary, spoke_repo = spoke_repo))
 
+            for alias_name, target in sorted(annotation.extra_aliased_targets.items()):
+                hub_contents.append("""
+alias(
+    name = "{alias_name}-{version}",
+    actual = "@{spoke_repo}//:{target}",
+)""".format(
+                    alias_name = alias_name,
+                    version = version,
+                    target = target,
+                    spoke_repo = spoke_repo,
+                ))
+
         workspace_versions = workspace_dep_versions_by_name.get(name)
         if workspace_versions:
             fq = sorted(workspace_versions)[-1]
+            workspace_version = fq.removeprefix(name + "-")
+            annotation = annotation_for(annotations, name, workspace_version)
 
             hub_contents.append("""
 alias(
@@ -844,13 +857,23 @@
     actual = ":{fq}",
 )""".format(name = name, fq = fq))
 
-            for binary in binaries:
+            for binary in annotation.gen_binaries:
                 hub_contents.append("""
 alias(
     name = "{name}__{binary}",
     actual = ":{fq}__{binary}",
 )""".format(name = name, fq = fq, binary = binary))
 
+            for alias_name in sorted(annotation.extra_aliased_targets.keys()):
+                hub_contents.append("""
+alias(
+    name = "{alias_name}",
+    actual = ":{alias_name}-{workspace_version}",
+)""".format(
+                    alias_name = alias_name,
+                    workspace_version = workspace_version,
+                ))
+
     workspace_deps, conditional_workspace_deps = render_select(
         [],
         workspace_dep_labels_by_triple,
@@ -1320,9 +1343,9 @@
         # "disable_pipelining": attr.bool(
         #     doc = "If True, disables pipelining for library targets for this crate.",
         # ),
-        # "extra_aliased_targets": attr.string_dict(
-        #     doc = "A list of targets to add to the generated aliases in the root crate_universe repository.",
-        # ),
+        "extra_aliased_targets": attr.string_dict(
+            doc = "A list of targets to add to the generated aliases in the root crate repository.",
+        ),
         # "gen_all_binaries": attr.bool(
         #     doc = "If true, generates `rust_binary` targets for all of the crates bins",
         # ),
diff -urN a/rs/private/annotations.bzl b/rs/private/annotations.bzl
--- a/rs/private/annotations.bzl	2026-03-21 22:53:52
+++ b/rs/private/annotations.bzl	2026-03-21 22:53:58
@@ -16,6 +16,7 @@
         crate_features = [],
         crate_features_select = {},
         gen_binaries = [],
+        extra_aliased_targets = {},
         rustc_flags = [],
         rustc_flags_select = {},
         patch_args = [],
@@ -41,6 +42,7 @@
         crate_features = crate_features,
         crate_features_select = crate_features_select,
         gen_binaries = gen_binaries,
+        extra_aliased_targets = extra_aliased_targets,
         rustc_flags = rustc_flags,
         rustc_flags_select = rustc_flags_select,
         patch_args = patch_args,
